testsuite/json-formatter.cc

```diff
@@ -0,0 +1,116 @@
+
+#define BOOST_TEST_DYN_LINK
+#define BOOST_TEST_MODULE snapper
+
+#include <boost/test/unit_test.hpp>
+
+#include <vector>
+#include <string>
+
+#include "../client/utils/JsonFormatter.h"
+
+using namespace std;
+
+BOOST_AUTO_TEST_CASE(test1_escape_values)
+{
+    snapper::cli::JsonFormatter::Data data = {
+	{ "key1", "value1" },
+	{ "key2", "value\"2" },
+	{ "key3", "value\\3" },
+	{ "key4", "\"value4\"" }
+    };
+
+    string expected_result =
+	"{\n"
+	"  \"key1\": \"value1\",\n"
+	"  \"key2\": \"value\\\"2\",\n"
+	"  \"key3\": \"value\\\\3\",\n"
+	"  \"key4\": \"\\\"value4\\\"\"\n"
+	"}";
+
+    snapper::cli::JsonFormatter formatter(data);
+
+    BOOST_CHECK_EQUAL(formatter.output(), expected_result);
+}
+
+
+BOOST_AUTO_TEST_CASE(test2_skip_format)
+{
+    snapper::cli::JsonFormatter::Data data = {
+	{ "key1", "value1" },
+	{ "key2", "value2" }
+    };
+
+    string expected_result =
+	"{\n"
+	"  \"key1\": value1,\n"
```

Comment:
`value1` is not a valid JSON value. I *guess* this is useful for boolean `true` and `false` but this should be expressed better in the tests.